### PR TITLE
feat(plugin-chart-choropleth-map): add zooming

### PR DESCRIPTION
**Pull request by kristw**
**NOTE: This is merging into a feature branch, not `master`, and will have subsequent PRs into the feature branch until completion.**

## 🏆 Enhancements

* Create `plugin-chart-choropleth-map` to replace the `country-map` and `world-map` charts.

### New features

* Allow zooming and panning with mini-map

### Coming in later PRs

* Encoding
* Tooltip

### Screenshots

image
image

### CODE REVIEW

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx [hunk 1]
```diff
@@ -92,11 +125,91 @@ export default class ChoroplethMap extends React.PureComponent<
...
+                Reset
+              </TextButton>
+            </ZoomControls>
+            <MiniMapControl>
+              <TextButton
+                type="button"
+                // eslint-disable-next-line react/jsx-handler-names
+                onClick={this.toggleMiniMap}
+              >
+                {showMiniMap ? 'Hide' : 'Show'} Mini Map
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx:207]: Add i18n for the copy?

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx [hunk 2]
```diff
@@ -92,11 +125,91 @@ export default class ChoroplethMap extends React.PureComponent<
 
   render() {
     const { height, width } = this.props;
+    const { showMiniMap } = this.state;
+
+    const renderedMap = this.renderMap();
+    const miniMapTransform = `translate(${(width * 3) / 4 - PADDING}, ${
+      (height * 3) / 4 - PADDING
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx:132]: Parentheses not needed

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx [hunk 3]
```diff
@@ -92,11 +125,91 @@ export default class ChoroplethMap extends React.PureComponent<
...
+    }) scale(0.25)`;
 
     return (
-      <svg width={width} height={height}>
-        {this.renderMap()}
-      </svg>
+      <Zoom
+        style={{ width, height }}
+        width={width}
+        height={height}
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx:139]: Is it required to have both width/height prop and style width/height for vx/zoom to work?

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx (hunk 2 above)
Comment (kristw) [reply plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx:132]: `eslint` added automatically

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx (hunk 3 above)
Comment (kristw) [reply plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx:139]: Yes, probably can file an issue to `@vx/zoom` too. The `width` and `height` attributes are used to compute the zoom transform, but it does not set the container (`<div>`) size correctly, had to specify style for that.

File: plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx (hunk 1 above)
Comment (kristw) [reply plugins/plugin-chart-choropleth-map/src/chart/ChoroplethMap.tsx]: done

**Pull request merged**